support listing loadlib and clear pass error

tiantn opened the pull request:
Signed-off-by: tian na <tiantn@cn.ibm.com>

dkelosky commented:
It looks like this is targeting #81 and #85.  

#85 seems fixed; here is the new error:
```
Command Error:
Username or password are not valid or expired.
```

dkelosky commented on src/FTPBase.Handler.ts:
```diff
@@ -36,7 +36,12 @@ export abstract class FTPBaseHandler implements ICommandHandler {
             await this.processFTP(additionalParameters);
         } catch (e) {
             this.log.error("Error encountered in FTP command:\n%s", require("util").inspect(e));
+            if (e.message.indexOf("PASS command failed") !== -1) {
```
these code changes make sense for #85 but I don't see any code changes that address #81.  listing a loadlib data set I still get:
```
Command Error:
Unrecognized file list header:  Name      Size     TTR   Alias-of AC --------- Attributes --------- Amode Rmode
```

tiantn replied:
Thank you @dkelosky, to support listing loadlib dataset members, the main code change is in zos-node-accessor, and I just update the zos-node-accesor’s version in package.json. Now it is 1.0.9 and you may need run npm install again to update the modules.

zFernand0 replied:
Shouldn't we make the regular dependencies static?
Meaning no `^` on `package.json#dependencies`
https://github.com/zowe/zowe-cli-ftp-plugin/blob/d7f541a493c24dfd1ea6bf2fbb1acee3608e5224/package.json#L36-L38

std4lqi replied:
Agree to use static version number in dependencies, to avoid the potential npm subtitution attacks.

tiantn replied:
Thanks @zFernand0 @std4lqi ! I will update it. Thank you very much!

lauren-li reviewed (commented):
@tiantn I think this looks good! I was able to manually list loadlibs, and the zftp error for the incorrect/missing credentials is also much clearer. All unit tests are passing.

The only thing left is a system test for listing loadlibs that keeps failing:
<img width="792" alt="Screen Shot 2021-07-26 at 6 14 28 PM" src="https://user-images.githubusercontent.com/45975633/127066652-e84249c0-0002-41f8-aacd-e8694fad303e.png">
This might have to do with the way my environment or test profile data is set up. @tiantn if you have any pointers for this, I would greatly appreciate it! Thank you!

tiantn commented:
Hi @std4lqi, @zFernand0, I updated the zos-node-accessor version to static. Thank you! And @lauren-li , thank you for your test, I found I missed the dsnLoadLib in default_properties. Now I added it and you just need input the loadlib dataset name in the custom_properties to run the system test. Thank you very much!

lauren-li commented:
> Hi @std4lqi, @zFernand0, I updated the zos-node-accessor version to static. Thank you! And @lauren-li , thank you for your test, I found I missed the dsnLoadLib in default_properties. Now I added it and you just need input the loadlib dataset name in the custom_properties to run the system test. Thank you very much!

@tiantn, I added the new loadlib dataset name into my custom_properties and re-ran the system tests. All looks good now. Thank you for the update!

zFernand0 commented:
I didn't get to test the functionality, but the code changes LGTM! 😋
